app/components/Header/DropdownContents/NotificationsDropdown/index.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+import cx from 'classnames';
+import { activityRenderers } from 'app/components/Feed';
+import Time from 'app/components/Time';
+import { Link } from 'react-router-dom';
+import ErrorBoundary from 'app/components/ErrorBoundary';
+import { toSpan } from 'app/components/Feed/context';
+import styles from './NotificationsDropdown.css';
+import shared from 'app/components/Header/Header.css';
+
+type Props = {
+  notificationsData: Object,
+  fetchNotifications: () => void,
+  notifications: Array<Object>,
+  markAllNotifications: () => Promise<void>,
+};
+
+type State = {
+  alreadyFetchedNotifications: boolean,
+};
```

Comment:
Can't see this used anywhere?

Reply:
Nope, but I plan on using it in order for notifications to not be fetched unnecessarily, e.g. when accidentally hovering over the notification icon